File: python/hash/포켓몬.py
"""
https://school.programmers.co.kr/learn/courses/30/lessons/1845
"""
def solution(nums):
    getNum = len(nums) / 2
    result = list(dict.fromkeys(nums))
    answer = getNum if len(result) > getNum else len(result)
    
    return answer

File: js/포켓몬.js
// https://school.programmers.co.kr/learn/courses/30/lessons/1845?language=javascript

function solution(nums) {
  let getNum = nums.length / 2;
  let result = [...new Set(nums)]; // 중복을 제거하여 폰켓몬의 종류만 남깁니다.
  let answer = result.length > getNum ? getNum : result.length;
  return answer;
}
